source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -355,8 +446,16 @@ class Filter : public Http::StreamFilter, Logger::Loggable<Logger::Id::lua> {
   }
   Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap& headers,
                                           bool end_stream) override {
+    const auto config_per_route = getConfigPerRoute(decoder_callbacks_.callbacks_);
+    PerLuaCodeSetup* setup = getPerLuaCodeSetup(config_.get(), config_per_route);
+    const int function_ref = setup ? setup->responseFunctionRef() : LUA_REFNIL;
+    if (function_ref == LUA_REFNIL) {
+      return Http::FilterHeadersStatus::Continue;
+    }
```

Comment:
This is copied from above. Can you share this in a utility?

Reply:
Okay, there is indeed some redundancy. I will find ways to reduce redundant code.